tcl/pd-gui.tcl

```diff
@@ -233,6 +233,28 @@ array set redo_actions {}
 set undo_action no
 set redo_action no
 
+# color scheme
+array set ::pd_colors {
+canvas_fill "white"
+gopbox_color "#FF8080"
+obj_txt_color "#000"
+msg_txt_color "#000"
+comment_color "#000"
+select_color "#00f"
+dash_outline "#f00"
+obj_box_outline "#ccc"
+msg_box_outline "#ccc"
+msg_box_fill ""
+obj_box_fill ""
+signal_cord "#558"
+msg_cord "#121"
+msg_nlet "#fff"
+graph_outline "#000"
+signal_nlet "#558"
+selrect_color "#000"
+txt_hilight "#37e744"
```

Comment:
Should this be "txt_highlight" ?

Reply:
could be.. I forget where I got these names from, but signal_nlet could also be signal_inlet

Reply:
Maybe it's nitpick, but these are the kinds of details that get fixed *in stone* once they go in... :)

Reply:
true, though maybe names like msg_box_fill should remain (instead of message_box_fill?).. not sure about selrect_color either (the color of the selection box for clicking & dragging in edit mode)

Reply:
I think the names originally come from extended, so if people still have color configurations from way back they would be back-compatible. But it's been so long now I don't know if that should be a consideration

Reply:
What do you think @umlaeute?

Reply:
I think "_nlet" is not a replacement for "_inlet", but is more an equivalent to something like "_xlet" (inlet or outlet).

Reply:
- i would prefer consistency over compatibility with Pd-extended
- i agree with @danomatika that we should use proper spelling
- i think that `msg_` is a common enough prefix (no need for `message_`)
- use either `_xlet` or `_iolet` (but not `_nlet`)
- drop any `_color` suffix (e.g. `comment_color`) it's totally redundant since those vars all live in the `::pd_colors` array anyhow
- what's `txt_highlight` anyhow (as opposed to e.g. `msg_txt` and `obj_txt`)? shouldn't that be something like `msg_txt_highlight` instead?
- same for `dash_outline`, shouldn't that be `obj_box_outline_dash`; or even `obj_box_outline_broken` (that is: use a semantic approach. we want to specify the color for "not-created objects", rather than for "dashed lines")

Reply:
@umlaeute  sounds good to me, txt_highlight is the color that text is highlighted in. I wasn't sure what default to set it to so I went with the osx green system highlight but thinking now maybe blue is better. Right now it gets set to whatever the system highlight is when pd starts (on osx at least).

Reply:
while i still don't understand what `txt_highlight` exactly means (is it the background colour of the selected text within an objbox if you click it?), t's definitely not *green* on my system (Debian).

if "Right now tcl/tk uses whatever the system highlight color is when pd starts", then i think this should stay the default.

Reply:
hmm I couldn't figure out how to get the system color from tcl so I guess I'll add a check to see if it's set and only pass the option to canvas if it is..

I don't think adding more cross-platform consistency in pd's behavior/appearance would be a negative though

Reply:
If `txt_hilight` is using the Tk `systemHighlight` or `systemHighlightText` colors, those are macOS-only, duding from [this](https://www.tcl.tk/man/tcl8.5/TkCmd/colors.htm).

Reply:
yes, it seems that systemHighlightText is also available on Winiows. I'm trying to peruse/decrypt the tk source to figure out what they do for other systems

Reply:
seems like they just use gray (#C3C3C3)